packaging/snapd.mk

```diff
@@ -0,0 +1,175 @@
+# This makefile is designed to be invoked from a build tree, not from the
+# source tree. This is particularly suited for building RPM packages. The goal
+# of this makefile is to *reduce* the duplication, *avoid* unintended
+# inconsistency between packaging of snapd across distributions.
+#
+# -----------------------------------------------------
+# Interface between snapd.mk and distribution packaging
+# -----------------------------------------------------
+#
+# Distribution packaging must generate snapd.defines.mk with a set of makefile
+# variable definitions that are discussed below. This allows the packaging
+# world to define directory layout and build configuration in one place and
+# this makefile to simply obey and implement that configuration.
+
+include snapd.defines.mk
+
+# There are two sets of definitions expected:
+# 1) variables defining various directory names
+vars += bindir sbindir libexecdir mandir datadir localstatedir unitdir
+# 2) variables defining build options:
+#   with_test_keys: set to 1 to build snapd with test key built in
+#   with_apparmor: set to 1 to build snapd with apparmor support
+#   with_core_bits: set to 1 to build snapd with things needed for the core/snapd snap
+#   with_alt_snap_mount_dir: set to 1 to build snapd with alternate snap mount directory
+vars += with_test_keys with_apparmor with_core_bits with_alt_snap_mount_dir
+# Verify that none of the variables are empty. This may happen if snapd.mk and
+# distribution packaging generating snapd.defines.mk get out of sync.
+
+$(foreach var,$(vars),$(if $(value $(var)),,$(error $(var) is empty or unset, check snapd.defines.mk)))
+
+# ------------------------------------------------
+# There are no more control knobs after this point
+# ------------------------------------------------
+
+# Import path of snapd.
+import_path = github.com/snapcore/snapd
+
+
+# This is usually set by %make_install. It is defined here to avoid warnings or
+# errors from referencing undefined variables.
+DESTDIR? =
+
+# Decide which of the two snap mount directories to use. This is not
+# referencing localstatedir because the code copes with only those two values
+# explicitly.
+ifeq ($(with_alt_snap_mount_dir),1)
+snap_mount_dir = /var/lib/snapd/snap
```

Comment:
Related to my earlier question - would it make sense to simply use a preset snap_mount_dir value here and get rid of with_alt_snap_mount_dir flag, with /snap as a default value?

Reply:
Answered above.